bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/editor/ClangdPlugin.java

```diff
@@ -48,14 +51,15 @@ public void start(BundleContext context) throws Exception {
 		workspaceTracker.open();
 		workspace = workspaceTracker.getService();
 		compileCommandsMonitor = new CompileCommandsMonitor(workspace).start();
-		cProjectChangeMonitor = new CProjectChangeMonitor().start();
+		cProjectChangeMonitor = Optional.ofNullable(PlatformUI.getWorkbench().getService(CProjectChangeMonitor.class))
```

Comment:
component may have methods annotated with
`@Activate` and `@Deactivate`, so we can avoid this code

Reply:
Thanks for the hint. I'll check that.

Reply:
Please take a look at my last commit